From 15de075af5943d18d4dc61ffc86d855f9fa39cc6 Mon Sep 17 00:00:00 2001
From: Steve Dickson <SteveD@redhat.com>
Date: Thu, 26 Jun 2014 15:33:33 -0400
Subject: [net] sunrpc: Don't let rpc_delay() clobber non-timeout errors

Message-id: <1403796815-28580-28-git-send-email-steved@redhat.com>
Patchwork-id: 84870
O-Subject: [RHEL7.1 PATCH 27/29] SUNRPC: Don't let rpc_delay() clobber non-timeout errors
Bugzilla: 1111170
RH-Acked-by: J. Bruce Fields <bfields@redhat.com>

From: Trond Myklebust <trond.myklebust@primarydata.com>

Signed-off-by: Trond Myklebust <trond.myklebust@primarydata.com>
Upsteam Id: 6bd144160a5554e4af052c153a094c4851a4c6aa
Bugzilla: 1111170
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/net/sunrpc/sched.c b/net/sunrpc/sched.c
index ff3cc4b..25578af 100644
--- a/net/sunrpc/sched.c
+++ b/net/sunrpc/sched.c
@@ -637,7 +637,8 @@ static void __rpc_queue_timer_fn(unsigned long ptr)
 
 static void __rpc_atrun(struct rpc_task *task)
 {
- task->tk_status = 0;
+ if (task->tk_status == -ETIMEDOUT)
+  task->tk_status = 0;
 }
 
 /*
-- 
1.7.1